contract/contracts/Elections/Elections.sol

```diff
@@ -0,0 +1,309 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.24;
+
+import {OwnableUpgradeable} from '@openzeppelin/contracts-upgradeable/access/OwnableUpgradeable.sol';
+import {PausableUpgradeable} from '@openzeppelin/contracts-upgradeable/utils/PausableUpgradeable.sol';
+import {Initializable} from '@openzeppelin/contracts-upgradeable/proxy/utils/Initializable.sol';
+import {ElectionTypes} from './ElectionTypes.sol';
+import {ElectionUtils} from './ElectionUtils.sol';
+
+/**
+ * @title Elections
+ * @dev Contract for managing Board of Directors elections with rank-order voting
+ * @notice This contract allows team owners to create elections, eligible voters to submit
+ *         rank-order votes, and automatically calculates results using modified Borda Count
+ */
+contract Elections is Initializable, OwnableUpgradeable, PausableUpgradeable {
+  // State variables
+  uint256 private _nextElectionId;
+  mapping(uint256 => ElectionTypes.Election) private _elections;
+  uint256[] private _electionIds;
+  mapping(uint256 => mapping(address => address[])) private _votes;
+
+  event ElectionCreated(
+    uint256 indexed electionId,
+    string title,
+    address indexed createdBy,
+    uint256 startDate,
+    uint256 endDate,
+    uint256 seatCount
+  );
+
+  event VoteSubmitted(uint256 indexed electionId, address indexed voter);
+
+  event ResultsPublished(uint256 indexed electionId, address[] winners);
+
+  // Custom errors
+  error ElectionNotFound();
+  error ElectionNotActive();
+  error ElectionEnded();
+  error AlreadyVoted();
+  error NotEligibleVoter();
+  error ResultsAlreadyPublished();
+  error ResultsNotReady();
+  error Unauthorized();
+
+  function initialize(address _owner) public initializer {
+    __Ownable_init(_owner);
+    __Pausable_init();
+    _nextElectionId = 1;
+  }
+
+  function pause() external onlyOwner {
+    _pause();
+  }
+
+  function unpause() external onlyOwner {
+    _unpause();
+  }
+
+  /**
+   * @dev Creates a new election
+   * @param title Election title
+   * @param description Election description
+   * @param startDate Election start timestamp
+   * @param endDate Election end timestamp
+   * @param seatCount Number of BOD seats (must be odd)
+   * @param candidates Array of candidates
+   * @param eligibleVoters Array of eligible voter addresses
+   * @return electionId The ID of the created election
+   */
+  function createElection(
+    string memory title,
+    string memory description,
+    uint256 startDate,
+    uint256 endDate,
+    uint256 seatCount,
+    address[] memory candidates,
+    address[] memory eligibleVoters
+  ) external onlyOwner whenNotPaused returns (uint256 electionId) {
+    ElectionUtils.validateSeatCount(seatCount);
```

Comment:
The owner can't create a new election when there is one current election. 
One current election can be: Ongoing, or Upcoming